app/components/versions-graph.js

```diff
@@ -0,0 +1,105 @@
+import semver from 'semver';
+import Component from '@ember/component';
+
+// Component to plot downloadeds split by semver minor version.
+export default Component.extend({
+  resizeHandler: undefined,
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    this.resizeHandler = () => this.rerender();
+    window.addEventListener('resize', this.resizeHandler, false);
+    document.addEventListener('googleChartsLoaded', this.resizeHandler, false);
+  },
+
+  willDestroyElement() {
+    window.removeEventListener('resize', this.resizeHandler);
+    document.removeEventListener('googleChartsLoaded', this.resizeHandler);
+  },
+
+  didRender() {
+    this._super(...arguments);
+
+    let data = this.data;
+
+    // Early exit if the google plotting libraries do not exist
+    let show = data && window.google && window.googleChartsLoaded;
+    this.element.style.display = show ? '' : 'none';
+    if (!show) {
+      console.error('cannot show graph due to missing google dependencies');
```

Comment:
In which situations would the dependencies not be present?

Reply:
When javascript is disabled.

Reply:
I presume you mean using NoScript selectively. The entire site does not work without Javascript, so we're not handling that use case. Again this is taking influence from `download-graph.js`, should the graph plotting scripts fail to download then we don't bother making the extra network requests.